otcextensions/tests/functional/osclient/nat/v2/test_dnat_rule.py

```diff
@@ -0,0 +1,199 @@
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+
+import json
+import uuid
+
+from openstackclient.tests.functional import base
+
+
+class TestDnatRule(base.TestCase):
+    """Functional Tests for NAT Gateway"""
+
+    UUID = uuid.uuid4().hex[:8]
+    ROUTER_NAME = 'sdk-test-router-' + UUID
+    NETWORK_NAME = 'sdk-test-net-' + UUID
+    SUBNET_NAME = 'sdk-test-subnet-' + UUID
+    PORT_NAME = 'sdk-test-port-' + UUID
+    ROUTER_ID = None
+    NETWORK_ID = None
+    FLOATING_IP_ID = None
+    PORT_ID = None
+
+    NAT_NAME = 'os-cli-test-' + UUID
+    NAT_ID = None
+    DNAT_RULE_ID = None
+
+    def test_01_nat_dnat_rule_list(self):
+        self.openstack(
+            'nat dnat rule list -f json '
+        )
+
+    def test_02_nat_dnat_rule_list_filters(self):
+        self.openstack(
+            'nat dnat rule list '
+            '--limit 1 --id 2 '
+            '--project-id 3 '
+            '--port-id 4 '
+            '--private-ip 5 '
+            '--internal-service-port 6 '
+            '--floating-ip-id 7 '
+            '--floating-ip-address 8 '
+            '--external-service-port 9 '
+            '--status 10 '
+            '--nat-gateway-id 11 '
+            '--protocol tcp '
+            '--admin-state-up true '
+        )
```

Comment:
at least some verification of response should be present

Reply:
validation check added